stockroom/external/importer/torchvision_importers.py

```diff
@@ -81,3 +86,327 @@ class FashionMnist(TorchvisionCommon):
     @classmethod
     def gen_splits(cls, root):
         return super().gen_splits(datasets.FashionMNIST, root)
+
+
+# TODO: make process_data consistant, and use counters
+class ImageFolder(BaseImporter):
+    name = "imagefolder"
+
+    def __init__(self, root):
+        root = Path(root)
+        try:
+            self.dataset = datasets.ImageFolder(root)
+        except FileNotFoundError:
+            raise RuntimeError(
+                f"Cannot find the directory '{os.path.expanduser(root)}'"
+            )
+
+        self.split = os.path.basename(root)
+        self.sample_img, self.sample_label = self._process_data(*self.dataset[0])
+
+        self._gen_metadata()
+        self._print_table()
+
+    def column_names(self):
+        return (
+            f"{self.name}-{self.split}-image",
+            f"{self.name}-{self.split}-label",
+        )
+
+    def shapes(self):
+        return self.shape
+
+    def dtypes(self):
+        return self.sample_img.dtype, self.sample_label.dtype
+
+    @classmethod
+    def gen_splits(cls, root):
+        return [cls(root)]
+
+    def __iter__(self):
+        for img, label in self.dataset:
+            yield self._process_data(img, label)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def variability_status(self):
+        return True, False
+
+    @staticmethod
+    def _process_data(img, lbl):
+        img = np.array(img)
+        lbl = np.array(lbl)
+        return img, lbl
+
+    def _gen_metadata(self):
+        print("Parsing data...")
+        H = W = 0
+        self.sample_counter = defaultdict(int)
+        for img, lbl in self.dataset:
+            # find max height/width
+            h, w = img.size
+            if h > H:
+                H = h
+            if w > W:
+                W = w
+
+            # find num samples in each class
+            self.sample_counter[lbl] += 1
+        _, _, c = self.sample_img.shape
+        self.shape = ((W, H, c), self.sample_label.shape)
```

Comment:
`_shapes` might be better?
```suggestion
        self._shapes = ((W, H, c), self.sample_label.shape)
```

Reply:
yeah